### PR TITLE
chore: clean up redirected links

### PR DESCRIPTION
**Pull request by nhcarrigan**
This PR updates the `301` links and removes the `307` links listed in #432.

### CODE REVIEW

**Comment by nhcarrigan** [comment]
It looks like the CI couldn't find the API token. 🤔

File: README.md [hunk 1]
```diff
@@ -43,25 +43,24 @@ If you have questions or comments, please create an issue.
...
 - [SourceSort](https://www.sourcesort.com) - searchable collection of open source projects looking for help, complete with metrics like first PR acceptance rate and response time that can make all the difference for new contributors.
 - [Issuehub.io](http://issuehub.io/) - a tool for searching GitHub issues by label and language.
 - [Code Triage](https://www.codetriage.com/) - another, really nice, tool for finding popular repositories and issues filtered by language.
 - [Awesome-for-beginners](https://github.com/MunGell/awesome-for-beginners) - a GitHub repo that amasses projects with good bugs for new contributors, and applies labels to describe them.
 - [Open Source Guides](https://opensource.guide/) - Collection of resources for individuals, communities, and companies who want to learn how to run and contribute to an Open Source project.
 - [45 Github Issues Dos and Don’ts](https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612) - Do's and Don'ts on GitHub.
-- [GitHub Guides](https://guides.github.com/) - basic guides on how to use GitHub effectively.
+- [GitHub Guides](https://guides.github.com/) - basic guides on how to use GitHub effectively.
```
Comment (erictleung) [inline README.md:54]: Why is this space added?

```suggestion
- [GitHub Guides](https://guides.github.com/) - basic guides on how to use GitHub effectively.
```

File: README.md [hunk 2]
```diff
@@ -81,26 +80,26 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
 - [Start Mozilla](https://twitter.com/StartMozilla) - a Twitter account that tweets about issues fit for contributors new to the Mozilla ecosystem.
 
 ## Useful articles for new Open Source contributors
 - [How to choose (and contribute to) your first Open Source project](https://github.com/collections/choosing-projects) by [@GitHub](https://github.com/github)
-- [How to find your first Open Source bug to fix](https://medium.freecodecamp.org/finding-your-first-open-source-project-or-bug-to-work-on-1712f651e5ba#.slc8i2h1l) by [@Shubheksha](https://github.com/Shubheksha)
-- [First Timers Only](https://kentcdodds.com/blog/first-timers-only) by [@kentcdodds](https://github.com/kentcdodds)
-- [Bring Kindness Back to Open Source](http://www.hanselman.com/blog/BringKindnessBackToOpenSource.aspx) by [@shanselman](https://github.com/shanselman)
+- [How to find your first Open Source bug to fix](https://www.freecodecamp.org/news/finding-your-first-open-source-project-or-bug-to-work-on-1712f651e5ba) by [@Shubheksha](https://github.com/Shubheksha)
+- [First Timers Only](https://kentcdodds.com/blog/first-timers-only) by [@kentcdodds](https://kentcdodds.com/blog/first-timers-only/)
```
Comment (erictleung) [inline README.md:91]: From the table in https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/432

  Line | Status | Link
| ---: | :----: | --- |
92 | [301](https://httpstatuses.com/301) | https://kentcdodds.com/blog/first-timers-only redirects to<br>https://kentcdodds.com/blog/first-timers-only/

points only to adding a forward slash to the end of the link itself. The GitHub user page was sufficient.

```suggestion
- [First Timers Only](https://kentcdodds.com/blog/first-timers-only/) by [@kentcdodds](https://github.com/kentcdodds)
```

File: README.md [hunk 3]
```diff
@@ -81,26 +80,26 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
+- [Bring Kindness Back to Open Source](https://www.hanselman.com/blog/BringKindnessBackToOpenSource.aspx) by [@shanselman](https://github.com/shanselman)
 - [Getting into Open Source for the First Time](https://www.nearform.com/blog/getting-into-open-source-for-the-first-time/) by [@mcdonnelldean](https://github.com/mcdonnelldean)
 - [How to Contribute to Open Source](https://opensource.guide/how-to-contribute/) by [@GitHub](https://github.com/github)
 - [How to Find a Bug in Your Code](https://8thlight.com/blog/doug-bradbury/2016/06/29/how-to-find-bug-in-your-code.html) by [@dougbradbury](https://twitter.com/dougbradbury)
-- [Mastering Markdown](https://guides.github.com/features/mastering-markdown/) by [@GitHubGuides](https://guides.github.com/)
+- [Mastering Markdown](https://guides.github.com/features/mastering-markdown/) by [@GitHubGuides](https://guides.github.com/)
```
Comment (erictleung) [inline README.md:96]: Not sure what this extra space is for.

```suggestion
- [Mastering Markdown](https://guides.github.com/features/mastering-markdown/) by [@GitHubGuides](https://guides.github.com/)
```

**Review by erictleung — CHANGES_REQUESTED** [review]
Looks good! Just a few minor changes.

File: README.md (hunk 1 above)
Comment (nhcarrigan) [reply README.md:54]: I'm not sure where that came from.

File: README.md (hunk 2 above)
Comment (nhcarrigan) [reply README.md:91]: Ah shoot I replace the wrong link. Thanks for the catch!

File: README.md [hunk 4]
```diff
@@ -109,35 +108,34 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
 - [Open Sources 2.0 : The Continuing Evolution](https://archive.org/details/opensources2.000diborich) (2005) - Open Sources 2.0 is a collection of insightful and thought-provoking essays from today's technology leaders that continues painting the evolutionary picture that developed in the 1999 book, Open Sources: Voices from the Revolution.
 - [The Architecture of Open Source Applications](http://www.aosabook.org/en/git.html) - Show how various aspects of Git work under the covers to enable distributed workflows, and how it differs from other version control systems (VCSs).
 - [Open Sources: Voices from the Open Source Revolution](https://www.oreilly.com/openbook/opensources/book/) - Essays from open-source pioneers such as Linus Torvalds (Linux), Larry Wall (Perl), and Richard Stallman (GNU).
 
 ## Open Source contribution initiatives
-- [Up For Grabs](http://up-for-grabs.net/#/) - Contains projects with beginner-friendly issues
+- [Up For Grabs](https://up-for-grabs.net/) - Contains projects with beginner-friendly issues
 - [First Timers Only](http://www.firsttimersonly.com/) - A list of bugs that are labelled "first-timers-only".
```
Comment (erictleung) [inline README.md:135]:
```suggestion
- [First Timers Only](https://www.firsttimersonly.com/) - A list of bugs that are labelled "first-timers-only".
```

File: README.md [hunk 5]
```diff
@@ -81,26 +80,26 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
 - [Mastering Markdown](https://guides.github.com/features/mastering-markdown/) by [@GitHubGuides](https://guides.github.com/)
 - [First mission: Contributors page](https://medium.com/@forCrowd/first-mission-contributors-page-df24e6e70705#.2v2g0no29) by [@forCrowd](https://github.com/forCrowd)
-- [How to make your first Open Source contribution in just 5 minutes](https://medium.freecodecamp.org/how-to-make-your-first-open-source-contribution-in-just-5-minutes-aaad1fc59c9a) by [@roshanjossey](https://medium.freecodecamp.org/@roshanjossey)
-- [Hacktoberfest 2019: How you can get your free shirt — even if you’re new to coding](https://medium.freecodecamp.org/hacktoberfest-2018-how-you-can-get-your-free-shirt-even-if-youre-new-to-coding-96080dd0b01b) by [@quincylarson](https://medium.freecodecamp.org/@quincylarson)
+- [How to make your first Open Source contribution in just 5 minutes](https://www.freecodecamp.org/news/how-to-make-your-first-open-source-contribution-in-just-5-minutes-aaad1fc59c9a) by [@roshanjossey](https://www.freecodecamp.org/news/@roshanjossey)
```
Comment (erictleung) [inline README.md:98]: Author profile on freeCodeCamp news doesn't exist, so let's just link to their GitHub page instead. Also, `awesome_bot` is picky with end forward slash.

```suggestion
- [How to make your first Open Source contribution in just 5 minutes](https://www.freecodecamp.org/news/how-to-make-your-first-open-source-contribution-in-just-5-minutes-aaad1fc59c9a/) by [@roshanjossey](https://github.com/Roshanjossey/)
```

File: README.md [hunk 6]
```diff
@@ -43,7 +43,7 @@ If you have questions or comments, please create an issue.
 - [License](#license)
 
 ## Contributing to Open Source in general
-- [The Definitive Guide to Contributing to Open Source](https://medium.freecodecamp.org/the-definitive-guide-to-contributing-to-open-source-900d5f9f2282) by [@DoomHammerNG](https://twitter.com/DoomHammerNG)
+- [The Definitive Guide to Contributing to Open Source](https://www.freecodecamp.org/news/the-definitive-guide-to-contributing-to-open-source-900d5f9f2282) by [@DoomHammerNG](https://twitter.com/DoomHammerNG)
```
Comment (erictleung) [inline README.md:46]: `awesome_bot` is being picky

```suggestion
- [The Definitive Guide to Contributing to Open Source](https://www.freecodecamp.org/news/the-definitive-guide-to-contributing-to-open-source-900d5f9f2282/) by [@DoomHammerNG](https://twitter.com/DoomHammerNG)
```

File: README.md [hunk 7]
```diff
@@ -43,7 +43,7 @@ If you have questions or comments, please create an issue.
...
 - [An Intro to Open Source](https://www.digitalocean.com/community/tutorial_series/an-introduction-to-open-source) - Tutorials by DigitalOcean to guide you on your way to contribution success here on GitHub.
 - [SourceSort](https://www.sourcesort.com) - searchable collection of open source projects looking for help, complete with metrics like first PR acceptance rate and response time that can make all the difference for new contributors.
```
Comment (erictleung) [inline README.md:48]: I reran the `awesome_bot` and it flagged this. I just checked and the site seems down. So I'd say to just remove this.

```suggestion
```

File: README.md [hunk 8]
```diff
@@ -81,26 +80,26 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
 - [Start Mozilla](https://twitter.com/StartMozilla) - a Twitter account that tweets about issues fit for contributors new to the Mozilla ecosystem.
 
 ## Useful articles for new Open Source contributors
 - [How to choose (and contribute to) your first Open Source project](https://github.com/collections/choosing-projects) by [@GitHub](https://github.com/github)
-- [How to find your first Open Source bug to fix](https://medium.freecodecamp.org/finding-your-first-open-source-project-or-bug-to-work-on-1712f651e5ba#.slc8i2h1l) by [@Shubheksha](https://github.com/Shubheksha)
-- [First Timers Only](https://kentcdodds.com/blog/first-timers-only) by [@kentcdodds](https://github.com/kentcdodds)
-- [Bring Kindness Back to Open Source](http://www.hanselman.com/blog/BringKindnessBackToOpenSource.aspx) by [@shanselman](https://github.com/shanselman)
+- [How to find your first Open Source bug to fix](https://www.freecodecamp.org/news/finding-your-first-open-source-project-or-bug-to-work-on-1712f651e5ba) by [@Shubheksha](https://github.com/Shubheksha)
```
Comment (erictleung) [inline README.md:90]:
```suggestion
- [How to find your first Open Source bug to fix](https://www.freecodecamp.org/news/finding-your-first-open-source-project-or-bug-to-work-on-1712f651e5ba/) by [@Shubheksha](https://github.com/Shubheksha)
```

File: README.md [hunk 9]
```diff
@@ -81,26 +80,26 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
-- [How to make your first Open Source contribution in just 5 minutes](https://medium.freecodecamp.org/how-to-make-your-first-open-source-contribution-in-just-5-minutes-aaad1fc59c9a) by [@roshanjossey](https://medium.freecodecamp.org/@roshanjossey)
-- [Hacktoberfest 2019: How you can get your free shirt — even if you’re new to coding](https://medium.freecodecamp.org/hacktoberfest-2018-how-you-can-get-your-free-shirt-even-if-youre-new-to-coding-96080dd0b01b) by [@quincylarson](https://medium.freecodecamp.org/@quincylarson)
+- [How to make your first Open Source contribution in just 5 minutes](https://www.freecodecamp.org/news/how-to-make-your-first-open-source-contribution-in-just-5-minutes-aaad1fc59c9a) by [@roshanjossey](https://www.freecodecamp.org/news/@roshanjossey)
+- [Hacktoberfest 2019: How you can get your free shirt — even if you’re new to coding](https://www.freecodecamp.org/news/hacktoberfest-2018-how-you-can-get-your-free-shirt-even-if-youre-new-to-coding-96080dd0b01b) by [@quincylarson](https://www.freecodecamp.org/news/@quincylarson)
```
Comment (erictleung) [inline README.md:99]:
```suggestion
- [Hacktoberfest 2019: How you can get your free shirt — even if you’re new to coding](https://www.freecodecamp.org/news/hacktoberfest-2018-how-you-can-get-your-free-shirt-even-if-youre-new-to-coding-96080dd0b01b/) by [@quincylarson](https://www.freecodecamp.org/news/author/quincylarson/)
```

File: README.md [hunk 10]
```diff
@@ -109,35 +108,34 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
 - [Oh shit, git!](https://ohshitgit.com/) - how to get out of common `git` mistakes described in plain English; also see [Dangit, git!](https://dangitgit.com/) for the page without swears.
-- [Atlassian Git Tutorials](https://www.atlassian.com/git/tutorials/) - various tutorials on using `git`.
+- [Atlassian Git Tutorials](https://www.atlassian.com/git/tutorials) - various tutorials on using `git`.
 - [GitHub Git Cheat Sheet](https://education.github.com/git-cheat-sheet-education.pdf) (PDF)
-- [freeCodeCamp's Wiki on Git Resources](https://www.freecodecamp.org/forum/t/wiki-git-resources/13136)
+- [freeCodeCamp's Wiki on Git Resources](https://forum.freecodecamp.org/t/wiki-git-resources/13136)
 - [GitHub Flow](https://www.youtube.com/watch?v=juLIxo42A_s) (42:06) - GitHub talk on how to make a pull request.
-- [GitHub Learning Resources](https://help.github.com/articles/git-and-github-learning-resources/) - Git and GitHub learning resources.
+- [GitHub Learning Resources](https://docs.github.com/articles/git-and-github-learning-resources/) - Git and GitHub learning resources.
```
Comment (erictleung) [inline README.md:115]: Weirdly enough, `awesome_bot` **doesn't** want a forward slash at the end.

```suggestion
- [GitHub Learning Resources](https://docs.github.com/articles/git-and-github-learning-resources) - Git and GitHub learning resources.
```

https://github.com/freeCodeCamp/how-to-contribute-to-open-source/pull/433/checks?check_run_id=1201626133

**Review by erictleung — CHANGES_REQUESTED** [review]
Just checked `awesome_bot`'s suggestions and made those suggestions in the comments. I made my changes based on the output from this https://github.com/freeCodeCamp/how-to-contribute-to-open-source/pull/433/checks?check_run_id=1201626133

**Comment by nhcarrigan** [comment]
Okay, looks like all the multi-redirects are fixed. 
Dupes will still show until the other PR is merged.

**Review by erictleung — APPROVED** [review]
LGTM! 👍 Thanks for cleaning these links up!

**Comment by nhcarrigan** [comment]
Thanks for your help! 😁
